tests/conformance_dsl/continuation.rs

```diff
@@ -230,6 +209,66 @@ pub(crate) struct EachBranch {
     fragment: Fragment,
 }
 
+#[derive(Clone, Debug, Default)]
+pub(crate) struct Produces {
+    pub elems: Vec<Element>,
+}
+
+impl Produces {
+    pub fn evaluate(&self, ctx: &Context) -> InnerResult<()> {
+        let (input, _encoding) = ctx.input(ctx.encoding())?;
+        let mut reader = ion_rs::Reader::new(ion_rs::AnyEncoding, input)?;
+
+        let mut is_equal = true;
+        let mut elem_iter = self.elems.iter();
+
+        while is_equal {
+            let (actual_value, expected_elem) = (reader.next()?, elem_iter.next());
+            match (actual_value, expected_elem) {
+                (None, None) => break,
+                (Some(actual_value), Some(expected_elem)) => 
+                    is_equal &= super::fragment::ProxyElement(expected_elem) == actual_value,
+                _ => is_equal = false,
+            }
+        }
+
+        if is_equal {
+            Ok(())
+        } else {
+            Err(ConformanceErrorKind::MismatchedProduce)
+        }
+    }
+}
+
+#[derive(Clone, Debug, Default)]
+pub(crate) struct Denotes {
+    pub model: Vec<ModelValue>,
+}
+
+impl Denotes {
+    pub fn evaluate(&self, ctx: &Context) -> InnerResult<()> {
+        let (input, _encoding) = ctx.input(ctx.encoding())?;
+        let mut reader = ion_rs::Reader::new(ion_rs::AnyEncoding, input)?;
+        let mut elem_iter = self.model.iter();
+
+        let mut is_equal = true;
+        while is_equal {
+            let (read_value, expected_element) = (reader.next()?, elem_iter.next());
+            match (read_value, expected_element) {
+                (Some(actual), Some(expected)) => is_equal = is_equal && (expected == actual),
+                (None, None) => break,
+                _ => is_equal = false,
+            }
```

Comment:
I think this is valid?
```suggestion
            is_equal = match (read_value, expected_element) {
                (Some(actual), Some(expected)) => is_equal && (expected == actual),
                (None, None) => break,
                _ => false,
            }
```